sdks/java/harness/src/test/java/org/apache/beam/fn/harness/FnApiDoFnRunnerTest.java

```diff
@@ -226,6 +226,7 @@ public void testUsingUserState() throws Exception {
             consumers,
             startFunctionRegistry,
             finishFunctionRegistry,
+            new ArrayList<>()::add,
```

Comment:
We could run teardown at the end of the test and ensure that no additional output has been added, ditto for the ones below.

Reply:
Sorry, I don't think I fully understand what you mean. Can you explain it more? :)

Reply:
after
```
    Iterables.getOnlyElement(finishFunctionRegistry.getFunctions()).run();
    assertThat(mainOutputValues, empty());
```
add something like:
```
    Iterables.getOnlyElement(tearDownFunctionRegistry.getFunctions()).run();
    assertThat(mainOutputValues, empty());
```